### ISSUE TITLE
Compact interactive stacked graphs representation?

### ISSUE THREAD

**Issue by nekohayo** [amit9838/mousam#130]
I find the current UI to be pretty space-inefficient in how it presents hourly and daily forecasts, as individual cells and rows. It takes two thirds of a full-HD screen to show those, and lots of scrolling.

I know this is a long shot "ponies on rainbows" enhancement request, but I find this app intriguing, as it does show more data than others I've tried (like GNOME Weather), but since this is the most advanced weather app I've found on the desktop, I thought I'd suggest this idea in case you might like it:

Any chance you could take a similar approach to WU's interactive charts, as seen in their "10 days" forecast tab ([example](https://www.wunderground.com/forecast/CYYZ))? It is pretty much the best weather UI I've ever seen, as it combines everything into a unified hourly+daily view of the next 10 days, and exact values are shown as the mouse hovers along the curves (note that the blue customize button also lets you show/hide any of the curves in those charts):

![image](https://github.com/amit9838/mousam/assets/479401/fde92f06-163c-4a2c-af17-54cb4590b763)

**Comment by amit9838** [comment]
OK, so here are my thoughts about this:
- The UI that you are referencing here is web based, And the rendering of web-based ui is different than native desktop UIs.
- I believe it is important to maintain the skeleton of native apps,  which is not an important constraint in web-based UIs.
- I also found that there not enough resource in gnome ecosystem for graph that can be used out of the box, and also in order the maintain the performance of the app, I've used cairo - native GUI toolkit, and generating the Bars from scratch, so some times it seems impossible to generate complex graph, although i had tried to generate the graph using cairo which was working good but still needs some refinements so i'll do it  in near future.
- I'll definitely add graphs, but for that i'll have to come up with new UI where i can add graph without making the UI cluttered.

**Comment by nekohayo** [comment]
Yes, I realize that web UIs have historically had fancier custom representations with no care for "fitting in a theme" because they don't have a standard widget library (well, other than Bootstrap) and the web is the Wild West :)

Cairo would let you make custom graphs on the desktop indeed, and I would suppose it would be sufficient? (there's Skia  if you want it to be GPU-accelerated, but I doubt Cairo would be a problem for a small not-constantly-animating graph like this). It might be possible to take inspiration from GNOME Weather's "Hourly" graph code, and GNOME System Monitor's CPU graphs code?

**Comment by amit9838** [comment]
Yes, i also like the graph in gnome weather. I'm trying to mimic the similar graph with some extra layers,  but these are not clearly planned yet.  I'll try this after 2-3 releases may be.
